a10_octavia/controller/worker/tasks/nat_pool_tasks.py

```diff
@@ -76,24 +76,25 @@ def execute(self, loadbalancer, vthunder, flavor_data=None):
 
     @axapi_client_decorator
     def revert(self, loadbalancer, vthunder, flavor_data=None, *args, **kwargs):
-        natpool_flavor_list = flavor_data.get('nat_pool_list')
-        natpool_flavor = flavor_data.get('nat_pool')
-        try:
-            if natpool_flavor_list:
-                for nat_pool in natpool_flavor_list:
-                    pool_name = nat_pool['pool_name']
+        if flavor_data:
+            natpool_flavor_list = flavor_data.get('nat_pool_list')
+            natpool_flavor = flavor_data.get('nat_pool')
+            try:
+                if natpool_flavor_list:
+                    for nat_pool in natpool_flavor_list:
+                        pool_name = nat_pool['pool_name']
+                        LOG.warning("Reverting creation of nat-pool: %s", pool_name)
+                        self.axapi_client.nat.pool.delete(pool_name)
+                if natpool_flavor:
+                    pool_name = natpool_flavor['pool_name']
                     LOG.warning("Reverting creation of nat-pool: %s", pool_name)
                     self.axapi_client.nat.pool.delete(pool_name)
-            if natpool_flavor:
-                pool_name = natpool_flavor['pool_name']
-                LOG.warning("Reverting creation of nat-pool: %s", pool_name)
-                self.axapi_client.nat.pool.delete(pool_name)
-        except ConnectionError:
-            LOG.exception(
-                "Failed to connect A10 Thunder device: %s", vthunder.ip_address)
-        except Exception as e:
-            LOG.exception("Failed to revert creation of the nat-pool due to: %s",
-                          str(e))
+            except ConnectionError:
```

Comment:
Apart from this, we won't be catching `ACOSException` also ?

Reply:
Checked in revert methods of other tasks. As `ACOSException`  is not catched in any of them, I haven't catched it here as well.

Reply:
Here we just log the exception now (maybe in the future we may have cases need ACOSException), so add ACOSException or not is fine for me.

Reply:
Adding it here